pnpm test rbac fail

Error: nexus-api#test: command (/home/runner/work/gdg-pup-platform/gdg-pup-platform/apps/nexus-api) /home/runner/setup-pnpm/node_modules/.bin/pnpm run test exited (1)
  4th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "RBAC CLI",
  ]

  5th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "",
  ]

  6th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "Options:",
  ]

  7th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  -h, --help      display help for command",
  ]

  8th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "",
  ]

  9th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "Commands:",
  ]

  10th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  roles           Manage RBAC roles.",
  ]

  11th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  users           Manage RBAC user assignments.",
  ]

  12th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  help [command]  display help for command",
  ]

  13th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "",
  ]

  14th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "Examples:",
  ]

  15th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  pnpm rbac roles create core-admin \"Core admin role\"",
  ]

  16th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  pnpm rbac roles permissions add core-admin roles mutations",
  ]

  17th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  pnpm rbac users roles add gdg-123 core-admin",
  ]

  18th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  pnpm rbac users permissions list gdg-123",
  ]


Number of calls: 18

 ✓ src/v1/modules/memberProjects/useCases/__tests__/MemberProjectImageUseCases.test.ts  (11 tests) 23ms
 ✓ src/v1/modules/MemberShowcase/__tests__/MemberShowcase.test.ts  (6 tests) 28ms
 ✓ src/v1/modules/gdgMerch/domain/__tests__/GdgMerch.test.ts  (12 tests) 25ms
 ✓ src/v1/modules/eventSystem/useCases/__tests__/SyncFromBevyRsvp.test.ts  (2 tests) 17ms
 ✓ src/v1/modules/bevyEvents/__tests__/ListBevyEvents.test.ts  (7 tests) 10ms
 ✓ src/v1/modules/pointsSystem/__tests__/ConsumePointsFromUser.test.ts  (9 tests) 12ms
 ✓ src/v1/modules/eventSystem/__tests__/EventSystemController.test.ts  (3 tests) 9ms
 ✓ src/v1/modules/pointsSystem/__tests__/Wallet.test.ts  (12 tests) 10ms
 ✓ src/v1/modules/memberTasks/__tests__/CompleteTask.test.ts  (6 tests) 8ms
 ✓ src/v1/modules/learningResources/useCases/__tests__/UpdateLearningResource.test.ts  (4 tests) 12ms
 ✓ src/v1/routes/gdgmembers/__tests__/gdgmembers.router.test.ts  (3 tests) 55ms
 ✓ src/v0/modules/filesModule/__tests__/ListFIlesWithPagination.test.ts  (3 tests) 7ms
 ✓ src/v1/modules/pointsSystem/__tests__/GivePointsToUser.test.ts  (8 tests) 14ms
 ✓ src/v1/modules/__deprecated_modules__/portfolioModule_deprecated/__tests__/UpdatePortfolioPropertyUseCase.test.ts  (7 tests) 7ms
 ✓ src/v1/modules/gdgMerch/useCases/__tests__/RedeemGdgMerch.test.ts  (4 tests) 12ms
 ✓ src/v1/modules/memberTasks/__tests__/CreateSystemTask.test.ts  (8 tests) 8ms
 ✓ src/v0/modules/pointsSystem/__tests__/GivePointsToUser.test.ts  (3 tests) 5ms
 ✓ src/v1/modules/learningResources/useCases/__tests__/ListLearningResources.test.ts  (5 tests) 9ms
 ✓ src/v0/modules/pointsSystem/__tests__/TakePointsFromUser.test.ts  (3 tests) 6ms
 ✓ src/v1/modules/memberTasks/__tests__/ListUserTasks.test.ts  (7 tests) 10ms
 ✓ src/v1/modules/memberTasks/__tests__/UpdateTask.test.ts  (6 tests) 6ms
 ✓ src/v1/modules/learningResources/__tests__/LearningResourceController.test.ts  (2 tests) 4ms
 ✓ src/v0/modules/pointsSystem/__tests__/ListUserTransactions.test.ts  (3 tests) 6ms
 ✓ src/v1/modules/members/domain/__tests__/GdgMember.listFieldsLimit.test.ts  (6 tests) 12ms
 ✓ src/v1/modules/learningResources/useCases/__tests__/CreateLearningResource.test.ts  (3 tests) 8ms
 ❯ src/v1/routes/events/__tests__/events.router.test.ts  (3 tests | 3 failed) 54ms
   ❯ src/v1/routes/events/__tests__/events.router.test.ts > EventsRouter sync routes > POST /events/:eventId/syncToBevy returns synced event including rsvp
     → expected 200 "OK", got 401 "Unauthorized"
   ❯ src/v1/routes/events/__tests__/events.router.test.ts > EventsRouter sync routes > POST /events/:eventId/syncToBevy returns a 400 for unlinked events
     → expected 400 "Bad Request", got 401 "Unauthorized"
   ❯ src/v1/routes/events/__tests__/events.router.test.ts > EventsRouter sync routes > POST /events/syncAllToBevy returns sync summary
     → expected 200 "OK", got 401 "Unauthorized"
 ✓ src/v0/modules/pointsSystem/__tests__/GetOneTransaction.test.ts  (3 tests) 7ms
 ✓ src/v0/modules/pointsSystem/__tests__/GetUserWallet.test.ts  (3 tests) 4ms
 ✓ src/v1/modules/pointsSystem/__tests__/ListUserTransactions.test.ts  (5 tests) 9ms
 ✓ src/v0/modules/filesModule/__tests__/GetOneFileById.test.ts  (2 tests) 14ms
 ✓ src/v1/modules/gdgMerch/useCases/__tests__/UpdateGdgMerchInfo.test.ts  (3 tests) 10ms
 ✓ src/v0/modules/filesModule/__tests__/UpdateFileById.test.ts  (2 tests) 8ms
 ✓ src/v1/modules/oneTimePin/__tests__/OneTimePin.test.ts  (3 tests) 9ms
 ✓ src/v1/modules/learningResources/domain/__tests__/LearningResource.test.ts  (3 tests) 18ms
 ✓ src/v1/modules/mailer/__tests__/SendEmail.test.ts  (4 tests) 15ms
stdout | src/v1/modules/mailer/__tests__/SendEmail.test.ts > SendEmail Use Case > should send an email with valid details

==================================================
📧 [MOCK EMAIL SENT]
--------------------------------------------------
To:      recipient@example.com
Subject: Hello World
Message: This is a test message.
Sent At: 4/17/2026, 8:22:55 AM
==================================================


 ✓ src/v1/modules/gdgMerch/useCases/__tests__/RestockGdgMerch.test.ts  (3 tests) 12ms
 ✓ src/v1/modules/__deprecated_modules__/portfolioModule_deprecated/__tests__/GetPortfolioByGdgIdUseCase.test.ts  (4 tests) 6ms
 ✓ src/v1/modules/__deprecated_modules__/portfolioModule_deprecated/__tests__/ListPortfoliosUseCase.test.ts  (4 tests) 8ms
 ✓ src/v0/modules/filesModule/__tests__/UploadFile.test.ts  (1 test) 5ms
 ✓ src/v1/modules/teamsSystem/__tests__/AddTeamMember.test.ts  (2 tests) 9ms
 ✓ src/v0/modules/filesModule/__tests__/DeleteFileById.test.ts  (2 tests) 11ms
 ✓ src/v1/modules/gdgMerch/useCases/__tests__/ListGdgMerch.test.ts  (2 tests) 8ms
 ✓ src/v1/modules/learningResources/useCases/__tests__/DeleteLearningResource.test.ts  (1 test) 10ms
 ✓ src/v1/modules/__deprecated_modules__/portfolioModule_deprecated/__tests__/GetPortfolioByNameUseCase.test.ts  (3 tests) 5ms
 ✓ src/v1/modules/products/useCases/__tests__/ListProducts.test.ts  (2 tests) 9ms
 ✓ src/v1/modules/teamsSystem/__tests__/UpdateTeam.test.ts  (2 tests) 5ms
 ✓ src/v1/modules/products/useCases/__tests__/UpdateProduct.test.ts  (2 tests) 8ms
 ✓ src/v1/modules/products/useCases/__tests__/CreateProduct.test.ts  (1 test) 10ms
 ✓ src/v1/modules/memberTasks/__tests__/GetTaskById.test.ts  (2 tests) 6ms
 ✓ src/v1/modules/teamsSystem/__tests__/CreateTeam.test.ts  (2 tests) 11ms
 ✓ src/v1/modules/products/useCases/__tests__/DeleteProduct.test.ts  (2 tests) 7ms
 ✓ src/v1/modules/gdgMerch/useCases/__tests__/DeleteGdgMerch.test.ts  (2 tests) 8ms
 ✓ src/v1/modules/learningResources/useCases/__tests__/GetLearningResource.test.ts  (2 tests) 8ms
 ✓ src/v1/modules/studyJams/__tests__/CreateStudyJam.test.ts  (2 tests) 12ms
 ✓ src/v1/modules/studyJams/__tests__/ListStudyJams.test.ts  (2 tests) 6ms
 ✓ src/v1/modules/gdgMerch/useCases/__tests__/CreateGdgMerch.test.ts  (1 test) 9ms
 ✓ src/v1/modules/products/useCases/__tests__/GetProduct.test.ts  (2 tests) 6ms
 ✓ src/v1/modules/pointsSystem/__tests__/GetOneTransaction.test.ts  (2 tests) 4ms
 ✓ src/v1/modules/gdgMerch/useCases/__tests__/GetGdgMerch.test.ts  (2 tests) 8ms
 ✓ src/v1/modules/studyJams/__tests__/UpdateStudyJam.test.ts  (2 tests) 4ms
 ✓ src/v1/modules/mailer/__tests__/OtpEmailTemplate.test.ts  (3 tests) 6ms
 ✓ src/v1/modules/teamsSystem/__tests__/ListTeams.test.ts  (2 tests) 5ms
 ✓ src/v1/modules/studyJams/__tests__/DeleteStudyJam.test.ts  (2 tests) 4ms
 ✓ src/v1/modules/pointsSystem/__tests__/GetUserWallet.test.ts  (2 tests) 4ms
 ✓ src/v1/modules/mailer/__tests__/MailerController.test.ts  (1 test) 5ms
stdout | src/v1/modules/mailer/__tests__/MailerController.test.ts > MailerController > should successfully send an email through the controller

==================================================
📧 [MOCK EMAIL SENT]
--------------------------------------------------
To:      test@example.com
Subject: Test Subject
Message: Test Message
Sent At: 4/17/2026, 8:22:58 AM
==================================================


 ✓ src/v1/modules/studyJams/__tests__/GetOneStudyJam.test.ts  (2 tests) 3ms

⎯⎯⎯⎯⎯⎯⎯ Failed Tests 4 ⎯⎯⎯⎯⎯⎯⎯

 FAIL  src/v1/modules/rbacSystem/__tests__/rbacCli.test.ts > runRbacCli > prints actionable usage error for unknown scope
AssertionError: expected "spy" to be called with arguments: [ Array(1) ]

Received: 

  1st spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "error: unknown command 'unknown'",
  ]

  2nd spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
e: pnpm rbac [options] [command]",
  ]

  3rd spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "",
  ]

  4th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "RBAC CLI",
  ]

  5th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "",
  ]

  6th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "Options:",
  ]

  7th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  -h, --help      display help for command",
  ]

  8th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "",
  ]

  9th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "Commands:",
  ]

  10th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  roles           Manage RBAC roles.",
  ]

  11th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  users           Manage RBAC user assignments.",
  ]

  12th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  help [command]  display help for command",
  ]

  13th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "",
  ]

  14th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "Examples:",
  ]

  15th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  pnpm rbac roles create core-admin \"Core admin role\"",
  ]

  16th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  pnpm rbac roles permissions add core-admin roles mutations",
  ]

  17th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  pnpm rbac users roles add gdg-123 core-admin",
  ]

  18th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  pnpm rbac users permissions list gdg-123",
  ]


Number of calls: 18

 ❯ src/v1/modules/rbacSystem/__tests__/rbacCli.test.ts:97:19
     95| 
     96|     expect(code).toBe(1);
     97|     expect(error).toHaveBeenCalledWith(
       |                   ^
     98|       "Invalid command: Unknown scope 'unknown'.",
     99|     );


⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/4]⎯
Error: AssertionError: expected "spy" to be called with arguments: [ Array(1) ]

Received: 

  1st spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "error: unknown command 'unknown'",
  ]

  2nd spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
e: pnpm rbac [options] [command]",
  ]

  3rd spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "",
  ]

  4th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "RBAC CLI",
  ]

  5th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "",
  ]

  6th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "Options:",
  ]

  7th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  -h, --help      display help for command",
  ]

  8th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "",
  ]

  9th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "Commands:",
  ]

  10th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  roles           Manage RBAC roles.",
  ]

  11th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  users           Manage RBAC user assignments.",
  ]

  12th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  help [command]  display help for command",
  ]

  13th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "",
  ]

  14th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "Examples:",
  ]

  15th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  pnpm rbac roles create core-admin \"Core admin role\"",
  ]

  16th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  pnpm rbac roles permissions add core-admin roles mutations",
  ]

  17th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  pnpm rbac users roles add gdg-123 core-admin",
  ]

  18th spy call:

  Array [
-   "Invalid command: Unknown scope 'unknown'.",
+   "  pnpm rbac users permissions list gdg-123",
  ]


Number of calls: 18

 ❯ src/v1/modules/rbacSystem/__tests__/rbacCli.test.ts:97:19




Error: Error: expected 200 "OK", got 401 "Unauthorized"
 ❯ src/v1/routes/events/__tests__/events.router.test.ts:63:8
 ❯ Test._assertStatus ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:309:14
 ❯ ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:365:13
 ❯ Test._assertFunction ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:342:13
 ❯ Test.assert ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:195:23
 ❯ localAssert ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:138:14
 ❯ Server.<anonymous> ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:152:11
 ❯ Object.onceWrapper node:events:638:28
 ❯ Server.emit node:events:524:28
 ❯ emitCloseNT node:net:2344:8


 FAIL  src/v1/routes/events/__tests__/events.router.test.ts > EventsRouter sync routes > POST /events/:eventId/syncToBevy returns synced event including rsvp
Error: expected 200 "OK", got 401 "Unauthorized"
 ❯ src/v1/routes/events/__tests__/events.router.test.ts:63:8
     61|     const response = await request(app)
     62|       .post(`/events/${syncedEventPayload.id}/syncToBevy`)
     63|       .expect(200);
       |        ^
     64| 
     65|     expect(syncEventToBevy).toHaveBeenCalledWith(syncedEventPayload.id…
 ❯ Test._assertStatus ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:309:14
 ❯ ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:365:13
 ❯ Test._assertFunction ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:342:13
 ❯ Test.assert ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:195:23
 ❯ localAssert ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:138:14
 ❯ Server.<anonymous> ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:152:11
 ❯ Object.onceWrapper node:events:638:28
 ❯ Server.emit node:events:524:28
 ❯ emitCloseNT node:net:2344:8


Error: Error: expected 400 "Bad Request", got 401 "Unauthorized"
 ❯ src/v1/routes/events/__tests__/events.router.test.ts:78:62
 ❯ Test._assertStatus ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:309:14
 ❯ ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:365:13
 ❯ Test._assertFunction ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:342:13
 ❯ Test.assert ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:195:23
 ❯ localAssert ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:138:14
 ❯ Server.<anonymous> ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:152:11
 ❯ Object.onceWrapper node:events:638:28
 ❯ Server.emit node:events:524:28
 ❯ emitCloseNT node:net:2344:8


⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/4]⎯

 FAIL  src/v1/routes/events/__tests__/events.router.test.ts > EventsRouter sync routes > POST /events/:eventId/syncToBevy returns a 400 for unlinked events
Error: expected 400 "Bad Request", got 401 "Unauthorized"
 ❯ src/v1/routes/events/__tests__/events.router.test.ts:78:62
     76|     const eventId = crypto.randomUUID();
     77| 
     78|     await request(app).post(`/events/${eventId}/syncToBevy`).expect(40…
       |                                                              ^
     79| 
     80|     expect(syncEventToBevy).toHaveBeenCalledWith(eventId);
 ❯ Test._assertStatus ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:309:14
 ❯ ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:365:13
 ❯ Test._assertFunction ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:342:13
 ❯ Test.assert ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:195:23
 ❯ localAssert ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:138:14
 ❯ Server.<anonymous> ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:152:11
 ❯ Object.onceWrapper node:events:638:28
 ❯ Server.emit node:events:524:28
 ❯ emitCloseNT node:net:2344:8


Error: Error: expected 200 "OK", got 401 "Unauthorized"
 ❯ src/v1/routes/events/__tests__/events.router.test.ts:94:8
 ❯ Test._assertStatus ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:309:14
 ❯ ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:365:13
 ❯ Test._assertFunction ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:342:13
 ❯ Test.assert ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:195:23
 ❯ localAssert ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:138:14
 ❯ Server.<anonymous> ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:152:11
 ❯ Object.onceWrapper node:events:638:28
 ❯ Server.emit node:events:524:28
 ❯ emitCloseNT node:net:2344:8


⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[3/4]⎯

 FAIL  src/v1/routes/events/__tests__/events.router.test.ts > EventsRouter sync routes > POST /events/syncAllToBevy returns sync summary
Error: expected 200 "OK", got 401 "Unauthorized"
 ❯ src/v1/routes/events/__tests__/events.router.test.ts:94:8
     92|     const response = await request(app)
     93|       .post("/events/syncAllToBevy")
     94|       .expect(200);
       |        ^
     95| 
     96|     expect(importAndSyncAllToBevy).toHaveBeenCalledTimes(1);
 ❯ Test._assertStatus ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:309:14
 ❯ ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:365:13
 ❯ Test._assertFunction ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:342:13
 ❯ Test.assert ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:195:23
 ❯ localAssert ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:138:14
 ❯ Server.<anonymous> ../../node_modules/.pnpm/supertest@7.2.2/node_modules/supertest/lib/test.js:152:11
 ❯ Object.onceWrapper node:events:638:28
 ❯ Server.emit node:events:524:28
 ❯ emitCloseNT node:net:2344:8

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[4/4]⎯

 Test Files  2 failed | 65 passed (67)
      Tests  4 failed | 248 passed (252)
   Start at  08:22:48
   Duration  10.82s (transform 2.91s, setup 6ms, collect 7.67s, tests 817ms, environment 19ms, prepare 9.85s)

 ELIFECYCLE  Test failed. See above for more details.
Error:  command finished with error: command (/home/runner/work/gdg-pup-platform/gdg-pup-platform/apps/nexus-api) /home/runner/setup-pnpm/node_modules/.bin/pnpm run test exited (1)
 ERROR  run failed: command  exited (1)

 Tasks:    1 successful, 2 total
Cached:    0 cached, 2 total
  Time:    11.985s 
Failed:    nexus-api#test

Error: Process completed with exit code 1.

--- a/apps/nexus-api/src/v1/modules/rbacSystem/__tests__/rbacCli.test.ts
+++ b/apps/nexus-api/src/v1/modules/rbacSystem/__tests__/rbacCli.test.ts
@@ -95,9 +95,8 @@ describe("runRbacCli", () => {
 
     expect(code).toBe(1);
     expect(error).toHaveBeenCalledWith(
-      "Invalid command: Unknown scope 'unknown'.",
+      "error: unknown command 'unknown'",
     );
-    expect(error).toHaveBeenCalledWith("Example: pnpm rbac roles ... | users ...");
   });
 
   it("prints usage and exits 0 when help flag is provided", async () => {

--- a/apps/nexus-api/src/v1/routes/events/__tests__/events.router.test.ts
+++ b/apps/nexus-api/src/v1/routes/events/__tests__/events.router.test.ts
@@ -5,6 +5,10 @@ import { EventsHttpController } from "../events.controller";
 import { EventsRouter } from "../events.router";
 import { BadRequestError } from "@/v1/errors/HttpError";
 
+vi.mock("@/v1/middlewares/rbac.middleware", () => ({
+  requirePermissions: vi.fn(() => (req: any, res: any, next: any) => next()),
+}));
+
 describe("EventsRouter sync routes", () => {
   const syncEventToBevy = vi.fn();
   const importAndSyncAllToBevy = vi.fn();

